x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph_grouped_node_preview_panel/graph_grouped_node_preview_panel.test.tsx

```diff
@@ -278,6 +279,53 @@ describe('GraphGroupedNodePreviewPanel', () => {
       });
     });
 
+    describe('Remount Behavior', () => {
+      it('should always reset to page 1 on mount, even with valid pageIndex in localStorage (grouped-entities)', () => {
+        const entityItems = Array.from({ length: 50 }, (_, i) =>
+          createEntityItem({ id: `entity-${i}` })
+        );
+
+        // Simulate localStorage with valid pageIndex=1 (within bounds for 50 items with pageSize=20)
+        // Before fix: would load pageIndex=1 from localStorage
+        // After fix: always resets to pageIndex=0 on mount
+        localStorage.setItem(
+          GROUPED_PREVIEW_PAGINATION_SETTINGS_KEY,
+          JSON.stringify({ pageIndex: 1, pageSize: 20 })
+        );
+
+        render(<GraphGroupedNodePreviewPanel {...defaultProps} entityItems={entityItems} />);
+
+        // Should ALWAYS start at page 1 (pageIndex=0), regardless of localStorage
+        expect(screen.getByTestId(CONTENT_BODY_TEST_ID)).toBeInTheDocument();
+        expect(screen.getByLabelText(/Page 1/)).toBeInTheDocument();
```

Comment:
I would also assert the hook is called with the correct `pageIndex`, as you do in the test case below:

```suggestion
        expect(screen.getByLabelText(/Page 1/)).toBeInTheDocument();
        expect(mockUseFetchDocumentDetails).toHaveBeenCalledWith(
          expect.objectContaining({
            options: expect.objectContaining({ pageIndex: 0 }),
          })
        );
```